src/routes/pdf-fetcher/containers/PdfFetcherPage.js

```diff
@@ -30,9 +30,13 @@ class PdfFetcherPage extends React.Component {
   }
 
   componentWillMount () {
-    const category = this.props.categories.getCategoryByUrl(this.props.fetchUrl)
-    if (category) {
-      this.fetchUrl(category)
+    try {
+      const category = this.props.categories.getCategoryByUrl(this.props.fetchUrl)
```

Comment:
sometimes you have the try block and sometimes you dont with getCategoryByUrl 🤔

Reply:
I think getCategoryByUrl should now throw an exception.

Reply:
there is always a try catch block, except from tests (which have to be refactored) and the use inside CategoriesMapModel getAncestors(), where there shouldn't be an error since all parents are existing (we checked that in the categories endpoint)

Reply:
@klinzo ok but then you should not catch them. This is not expected and we're fine if the app crashes if the backend returns wrong data. @Schedulaar @NKlug What are your opinions?

Reply:
No the app should not crash: getCategoryByUrl would always return undefined, if the user navigates to a dead/invalid link where we should show an error.

Reply:
we could also remove the exception from getCategoryByUrl, but getAncestors will be ugly than (because of flow)